server/service/src/repack/insert.rs

```diff
@@ -0,0 +1,516 @@
+            .pop()
+            .unwrap();
+        assert_eq!(
+            service.insert_repack(
+                &context,
+                inline_init(|r: &mut InsertRepack| {
+                    r.stock_line_id = mock_stock_line_b().id.clone();
+                    r.number_of_packs = 40.0;
+                    r.new_pack_size = 2;
+                })
+            ),
+            Err(ServiceError::StockLineReducedBelowZero(stock_line))
+        );
+    }
+
+    #[actix_rt::test]
+    async fn insert_repack_success() {
+        let stock_line_a = StockLineRow {
+            id: "stock_line_a".to_string(),
+            item_id: "item_a".to_string(),
+            store_id: mock_store_a().id.clone(),
+            pack_size: 5,
+            cost_price_per_pack: 0.20,
+            sell_price_per_pack: 0.50,
+            available_number_of_packs: 100.0,
+            total_number_of_packs: 100.0,
+            ..Default::default()
+        };
+
+        let (_, connection, connection_manager, _) = setup_all_with_data(
+            "insert_repack_success",
+            MockDataInserts::all(),
+            inline_init(|r: &mut MockData| {
+                r.stock_lines = vec![stock_line_a.clone()];
+            }),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let context = service_provider
+            .context(mock_store_a().id, mock_user_account_a().id.to_string())
+            .unwrap();
+        let service = service_provider.repack_service;
+
+        let invoice_repo = InvoiceRowRepository::new(&connection);
+        let invoice_line_repo = InvoiceLineRowRepository::new(&connection);
+        let stock_line_repo = StockLineRowRepository::new(&connection);
+
+        // Repack increase where stock pack size == one
+        let increased_pack_size = service
+            .insert_repack(
+                &context,
+                inline_init(|r: &mut InsertRepack| {
+                    r.stock_line_id = mock_stock_line_a().id.clone();
+                    r.number_of_packs = 8.0;
+                    r.new_pack_size = 2;
+                }),
+            )
+            .unwrap();
+
+        let invoice = invoice_repo
+            .find_one_by_id(&increased_pack_size.invoice_row.id)
+            .unwrap();
+        let invoice_lines = invoice_line_repo
+            .find_many_by_invoice_id(&invoice.id)
+            .unwrap();
+
+        let stock_line_ids: Vec<String> = invoice_lines
+            .iter()
+            .map(|line| line.stock_line_id.clone().unwrap())
+            .collect();
+        let stock_lines = stock_line_repo.find_many_by_ids(&stock_line_ids).unwrap();
+
+        assert_eq!(invoice_lines.len(), 2);
+        // New stock line
+        assert_eq!(
+            stock_lines[0],
+            inline_init(|s: &mut StockLineRow| {
+                s.id = stock_lines[0].id.clone();
+                s.item_id = mock_stock_line_a().item_id.clone();
```

Comment:
inline_edit would change the record, so can't check if original stock line has been updated, no? 🤔

Reply:
`inline_edit` is badly named, it doesn't actually edit original record (otherwise it would need to be `mut`), it clones it and allows you to edit the clone. Anyway i think we should really use the `spread` in cases like this:

```rust
        assert_eq!(
            stock_lines[0],
            StockLineRow {
                id: stock_lines[0].id.clone(),
                available_number_of_packs: 4.0,
                total_number_of_packs: 4.0,
                pack_size: 2,
                cost_price_per_pack: mock_stock_line_a().cost_price_per_pack * 2.0,
                sell_price_per_pack: mock_stock_line_a().sell_price_per_pack * 2.0,
                // Matches all other fields
                ..mock_stock_line_a()
            }
        );
```